packages/block-editor/src/hooks/child-layout.js

```diff
@@ -41,7 +43,21 @@ export function ChildLayoutEdit( {
 } ) {
 	const { style = {} } = attributes;
 	const { layout: childLayout = {} } = style;
-	const { selfStretch, flexSize } = childLayout;
+	const { selfStretch, flexSize, breakOutOfLayout = false } = childLayout;
+
+	useEffect( () => {
```

Comment:
I remember we needed to remove the useEffect in the Group block back in https://github.com/WordPress/gutenberg/pull/44176. Will adding a `useEffect` here have similar performance concerns as the earlier Group block useEffect?

I'm wondering if we need to change the attribute via `setAttribute` or if it'd be possible to use a different temporary variable for determining which control to display, if it's more about which tab is shown when the block is initially selected, rather than changing the attributes for the block 🤔

Reply:
Offhand I'd think not because this control only loads once the block is selected. I think the perf issues we were seeing with using an effect in the Group edit function were related to the editor content taking longer to load. 

In any case, I'm happy to change if there's a better approach! I played around with it a bit but couldn't find another way to tell if the component is rendering for the first time, or just updating as a result of a change of value.

Reply:
> Offhand I'd think not because this control only loads once the block is selected.

Ah, good point, yes, the rendering is neatly tucked away inside the `DimensionsPanel` isn't it — sounds like we can go with this approach for now and revisit if need be 👍